projects/plugins/crm/api/create_transaction.php

```diff
@@ -27,19 +27,37 @@
 		$json_params = file_get_contents( 'php://input' );
 		$new_trans   = json_decode( $json_params, true );
 
-		// REQUIRED
-		$orderid = -1;
+// REQUIRED FIELDS
+$required_fields = array(
+	'orderid',
+	'email',
+);
+
+// Check for required fields. Fail if any are missing.
+foreach ( $required_fields as $field ) {
+	if ( ! array_key_exists( $field, $new_trans ) ) {
+		wp_send_json(
+			array(
+				'error'   => 400,
+				'message' => 'Missing required field: ' . $field,
+			)
+		);
+		exit();
+	}
+}
+
+$orderid = -1;
+$email   = '';
```

Comment:
In the current implementation (which I don't think we should merge) we wouldn't need these "default required" lines.

Reply:
I think you've misinterpreted the earlier comment. It's not required for the API call (user-side), but instead is required for the DAL call (server-side).

Reply:
?
It's required by the DAL, and here we add the check.

Note: Generally as a quick fix/improvement, made on the fly, I left the previous implementation and consideration, like the default values. In this case, I added the check at the API level ($email is required by the API to be able to get the contact id, required by the Transaction DAL)

Reply:
I added two more fields as required ones: `status` and `total`.

Reply:
And removed `email`. So `$email` and `$fname` will have a default value.